src/category_theory/limits/preserves/shapes/terminal.lean

```diff
@@ -114,4 +118,95 @@ begin
   apply_instance,
 end
 
+end terminal
+
+section initial
+
+/--
+The map of an empty cocone is a colimit iff the mapped object is initial.
+-/
+def is_colimit_map_cocone_empty_cocone_equiv :
+  is_colimit (G.map_cocone (as_empty_cocone X)) ≃ is_initial (G.obj X) :=
+(is_colimit.precompose_hom_equiv (functor.empty_ext _ _) _).symm.trans
+  (is_colimit.equiv_iso_colimit (cocones.ext (iso.refl _) (by tidy)))
+
+/-- The property of preserving initial objects expressed in terms of `is_initial`. -/
+def is_initial_obj_of_is_initial [preserves_colimit (functor.empty C) G]
+  (l : is_initial X) : is_initial (G.obj X) :=
+is_colimit_map_cocone_empty_cocone_equiv G X (preserves_colimit.preserves l)
+
+/-- The property of reflecting initial objects expressed in terms of `is_initial`. -/
+def is_initial_of_is_initial_obj [reflects_colimit (functor.empty C) G]
```

Comment:
It would be nice to use dot-notation here as well. Maybe the following name?
```suggestion
def is_initial.is_initial_of_obj [reflects_colimit (functor.empty C) G]
```